imap_processing/ena_maps/ena_maps.py

```diff
@@ -0,0 +1,543 @@
+        )
+
+    return flat_indices_input_grid_output_frame
+
+
+# Define the pointing set classes
+class PointingSet(ABC):
+    """
+    Abstract class to contain pointing set (PSET) data in the context of ENA sky maps.
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        Dataset containing the pointing set data.
+    spice_reference_frame : geometry.SpiceFrame
+        The reference Spice frame of the pointing set.
+    """
+
+    @abstractmethod
+    def __init__(self, dataset: xr.Dataset, spice_reference_frame: geometry.SpiceFrame):
+        """Abstract method to initialize the pointing set object."""
+        self.spice_reference_frame = spice_reference_frame
+        self.num_points = 0
+        self.az_el_points = np.zeros((self.num_points, 2))
+        self.data = xr.Dataset()
+
+    def __repr__(self) -> str:
+        """
+        Return a string representation of the pointing set.
+
+        Returns
+        -------
+        str
+            String representation of the pointing set.
+        """
+        return (
+            f"{self.__class__} PointingSet"
+            f"(spice_reference_frame={self.spice_reference_frame})"
+        )
+
+
+class UltraPointingSet(PointingSet):
+    """
+    PSET object specifically for ULTRA data, nominally at Level 1C.
+
+    Parameters
+    ----------
+    l1c_dataset : xr.Dataset | pathlib.Path | str
+        L1c xarray dataset containing the pointing set data or the path to the dataset.
+        Currently, the dataset is expected to be in a rectangular grid,
+        with data_vars indexed along the coordinates:
+            - 'epoch' : time value (1 value per PSET)
+            - 'azimuth_bin_center' : azimuth bin center values
+            - 'elevation_bin_center' : elevation bin center values
+        Some data_vars may additionally be indexed by energy bin;
+        however, only the spatial axes are used in this class.
+    spice_reference_frame : geometry.SpiceFrame
+        The reference Spice frame of the pointing set. Default is IMAP_DPS.
+
+    Raises
+    ------
+    ValueError
+        If the azimuth or elevation bin centers do not match the constructed grid.
+        Or if the azimuth or elevation bin spacing is not uniform.
+    ValueError
+        If multiple epochs are found in the dataset.
+    """
+
+    def __init__(
+        self,
+        l1c_dataset: xr.Dataset | pathlib.Path | str,
+        spice_reference_frame: geometry.SpiceFrame = geometry.SpiceFrame.IMAP_DPS,
+    ):
+        # Store the reference frame of the pointing set
+        self.spice_reference_frame = spice_reference_frame
+
+        # Read in the data and store the xarray dataset as data attr
+        if isinstance(l1c_dataset, (str, pathlib.Path)):
+            self.path: str | None = str(l1c_dataset)
+            self.data = xr.open_dataset(l1c_dataset)
+        elif isinstance(l1c_dataset, xr.Dataset):
+            self.path = None
+            self.data = l1c_dataset
```

Comment:
I don't think we need to store the path since you aren't using it later. Additionally, I thought `xr` couldn't open cdf files and we needed to use our cdflib helpers?

https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/d9e683ceacc69c83c07330d9a19de572b99baae6/imap_processing/cdf/utils.py#L21-L25

```suggestion
        if isinstance(l1c_dataset, (str, pathlib.Path)):
            self.data = load_cdf(l1c_dataset)
        elif isinstance(l1c_dataset, xr.Dataset):
            self.data = l1c_dataset
```

Reply:
Good catch, and thanks for the suggestion!